### ISSUE TITLE
Getting 0 success rate for kitchen environments

### ISSUE THREAD

**Issue by lakshitadodeja** [irom-princeton/dppo#34]
Hi, 

I have been trying to evaluate the performance of pretrained policies which works fine with other environments but I am getting 0 success rate with kitchen environment. I also created a new conda environment and only installed kitchen specific dependencies with `pip install -e .[kitchen]`  but the success rate was still 0. Here is my current config file (Also tried with mixed and partial) - 

```
defaults:
  - _self_
hydra:
  run:  
    dir: ${logdir}
_target_: agent.eval.eval_diffusion_agent.EvalDiffusionAgent

name: ${env_name}_eval_diffusion_mlp_ta${horizon_steps}_td${denoising_steps}
logdir: ${oc.env:DPPO_LOG_DIR}/gym-eval/${name}/${now:%Y-%m-%d}_${now:%H-%M-%S}_${seed}
base_policy_path: ${oc.env:DPPO_LOG_DIR}/gym-pretrain/kitchen-complete-v0_pre_diffusion_mlp_ta4_td20/2024-10-20_16-47-42_42/checkpoint/state_8000.pt
normalization_path: ${oc.env:DPPO_DATA_DIR}/gym/${env_name}/normalization.npz

seed: 42
device: cuda:0
env_name: kitchen-complete-v0
obs_dim: 60
action_dim: 9
denoising_steps: 20
cond_steps: 1
horizon_steps: 4
act_steps: 4

n_steps: 70
render_num: 0

env:
  n_envs: 40
  name: ${env_name}
  max_episode_steps: 280
  reset_at_iteration: False
  save_video: False
  best_reward_threshold_for_success: 4
  wrappers:
    mujoco_locomotion_lowdim:
      normalization_path: ${normalization_path}
    multi_step:
      n_obs_steps: ${cond_steps}
      n_action_steps: ${act_steps}
      max_episode_steps: ${env.max_episode_steps}
      reset_within_step: True

model:
  _target_: model.diffusion.diffusion.DiffusionModel
  predict_epsilon: True
  denoised_clip_value: 1.0
  #
  network_path: ${base_policy_path}
  network:
    _target_: model.diffusion.mlp_diffusion.DiffusionMLP
    time_dim: 16
    mlp_dims: [256, 256, 256]
    cond_mlp_dims: [128, 32]
    residual_style: True
    cond_dim: ${eval:'${obs_dim} * ${cond_steps}'}
    horizon_steps: ${horizon_steps}
    action_dim: ${action_dim}
  horizon_steps: ${horizon_steps}
  obs_dim: ${obs_dim}
  action_dim: ${action_dim}
  denoising_steps: ${denoising_steps}
  device: ${device}
```

**Comment by lakshitadodeja** [comment]
Initially I was getting this error -

**Comment by lakshitadodeja** [comment]
Sorry got marked completed accidentally. 
Initially I was getting this error - 

```
Error in call to target 'model.diffusion.diffusion_eval_ft.DiffusionEvalFT':
TypeError("init() missing 1 required positional argument: 'use_ddim'")
```

Then I added this in config - 
```
use_ddim: ${use_ddim}
 ddim_steps: ${ddim_steps}
```

Now I am getting this error - 
```
RuntimeError('Error(s) in loading state_dict for DiffusionMLP:\n\tMissing key(s) in state_dict: "time_embedding.1.weight", "time_embedding.1.bias", "time_embedding.3.weight", "time_embedding.3.bias", "cond_mlp.moduleList.0.linear_1.weight", "cond_mlp.moduleList.0.linear_1.bias", "cond_mlp.moduleList.1.linear_1.weight", "cond_mlp.moduleList.1.linear_1.bias", "mlp_mean.layers.0.weight", "mlp_mean.layers.0.bias", "mlp_mean.layers.1.l1.weight", "mlp_mean.layers.1.l1.bias", "mlp_mean.layers.1.l2.weight", "mlp_mean.layers.1.l2.bias", "mlp_mean.layers.2.weight", "mlp_mean.layers.2.bias". ')
full_key: model
```

Do I need to pretrain the models again ?

**Comment by allenzren** [comment]
Hey! Sorry there was an issue with the code. I just updated the branch (also merged with the main branch). Basically the eval config was not compatible with evaluating pre-trained (not fine-tuned) policy, which seems what you were trying?

You can use the main branch now for eval, either evaluating pre-trained or fine-tuned policy. Just make sure to specify `ft_denoising_steps` to 0 if evaluating pre-trained policy, and to the correct number of denoising steps fine-tuned if evaluating fine-tuned policy. I added a note on this in the evaluation section of the README.

Also, I realize the zero success rate is expected --- there is no issue with running evaluation with pre-trained policy in kitchen, in the original main branch. There are four stages in the kitchen task, and if the policy completes all four, it receives a reward of 4 (1 for each stage). The curves in Fig. 6 of the paper shows the normalized score (out of 4), instead of the success rate. For example, below shows the results I just ran with the pre-trained policy in Kitchen-Complete in the updated main branch, and it shows an average reward of 2.675, which is a normalized score of 2.675/4=0.66875.

```
[2025-02-04 11:57:59,841][root][INFO] - Number of network parameters: 175604
[2025-02-04 11:57:59,921][root][INFO] - Actor weights not found. Using pre-trained weights!
[2025-02-04 11:57:59,921][root][INFO] - Loaded base policy weights from /home/allen/dppo-dev/result/gym-pretrain/kitchen-complete-v0_pre_diffusion_mlp_ta4_td20/2024-10-20_16-47-42_42/checkpoint/state_8000.pt
Processed step 0 of 70
Processed step 10 of 70
Processed step 20 of 70
Processed step 30 of 70
Processed step 40 of 70
Processed step 50 of 70
Processed step 60 of 70
[2025-02-04 11:58:06,897][agent.eval.eval_diffusion_agent][INFO] - eval: num episode   40 | success rate   0.0000 | avg episode reward   2.6750 | avg best reward   0.2500
```

Hope this makes sense!

**Comment by lakshitadodeja** [comment]
It works now ! 
Thank you !!
